packages/regenerator-transform/src/visit.js

```diff
@@ -16,6 +16,53 @@ import * as util from "./util";
 
 let getMarkInfo = require("private").makeAccessor();
 
+// we want to convert this shorthand object method into a normal object property
+// which is a function expression. for example, this:
+//  var foo = {
+//    *bar(baz) { return 5; }
+//  }
+// should be replaced with:
+//  var foo = {
+//    bar: function*(baz) { return 5; }
+//  }
+// to do this, we will clone the parameter array and the body of the function
+// into a new FunctionExpression.
+function replaceShorthandObjectMethod(path) {
+  if (!t.isObjectMethod(path.node)) {
+    return path;
+  }
+
+  // if it's not a generator method, we don't need to do anything.
+  if (!path.node.generator) {
+    return path;
+  }
+
+  const parameters = path.node.params.map(function (param) {
+    return t.cloneDeep(param);
```

Comment:
How necessary are these deep clones (this one and the others)? Are we really worried about preserving the parts of the original AST that are being replaced?